pkg/rancher-desktop/main/update/LonghornProvider.ts

```diff
@@ -186,6 +206,79 @@ export async function setHasQueuedUpdate(isQueued: boolean): Promise<void> {
   }
 }
 
+async function getPlatformVersion(): Promise<string> {
+  switch (process.platform) {
+  case 'win32':
+    return os.release();
+  case 'darwin': {
+    const macOsVersion = await getMacOsVersion(console);
+
+    if (macOsVersion === null) {
+      throw new Error('failed to get macOS version');
+    }
+
+    return macOsVersion.toString();
+  }
+  case 'linux':
+    // OS version is hard to get on Linux. Luckily, automatic updates
+    // are not supported on Linux as of the time of writing. "unknown"
+    // should suffice.
+    return 'unknown';
+  }
+  throw new Error(`Platform "${ process.platform }" is not supported`);
+}
+
+/**
+ * Fetch info on available versions of Rancher Desktop, as well as other
+ * things, from the Upgrade Responder server.
+ */
+export async function queryUpgradeResponder(url: string, currentVersion: semver.SemVer): Promise<UpgradeResponderQueryResult> {
+  const requestPayload = {
+    appVersion: currentVersion,
+    extraInfo:  {
+      platform:        `${ process.platform }-${ os.arch() }`,
```

Comment:
I would suggest to provide the `platform` as separate fields instead of smashing them together (note that `process.platform` is a function, it is missing parentheses):

```suggestion
      platformName:    process.platform(),
      platformArch:    os.arch(),
```

We send this as JSON, so making the server parse `platform` to separate the values doesn't make any sense.

We can keep the existing `platform` field for backwards compatibility, even though it would now be redundant. But is `platform` even used in the current LonghornUpdateProvider server?

Reply:
I left this as is to preserve the form of any data the upgrade responder has gathered. IIRC we can use the keys in `extraInfo` to work with usage data, so if we care about this it would be easier if we didn't change this.

I'll leave it as is for now. Let me know if you're sure we want to change it.